src/Controller/Component/RedirectComponent.php

```diff
@@ -77,8 +77,10 @@ public function redirectToAuthenticationPage($url = null)
         // authentication failure flash messages from showing. The page will be
         // processed as per normal when the user is redirected after the token cookie
         // has been checked.
-        $resp->send();
-        $resp->stop();
+        if (!is_null($resp)) {
```

Comment:
I'm not sure about this. Perhaps the issue is that `RedirectComponent::redirectToAuthenticationPage` should not be called at all in this case? Is this not already solved by your changes in a40364d?

Reply:
Unfortunately a40364d does not solve this issue. The issue arises when a user has intercepted the request with beforeRedirect and then they stop propagation of the event. This causes the Controller to return null instead of a $response object that we can act upon. I'm not sure if we can determine this back CookieTokenAuthenticate::authenticate in order to not call RedirectComponent::redirectToAuthenticationPage in the first place or not.

I will look into it...

Reply:
Just to re-iterate, this issue would occur with or without it being an ajax call. That is why it's not fixed or associated with https://github.com/Beskhue/CookieTokenAuth/commit/a40364dd697ac7c44c7f4245a5fe3f3c28901e57.

Reply:
Alright. In that case the check makes sense, and in any case it won't hurt. I'll add a small comment that `$resp` can be null due to stopping of event propagation.